flow/aggregates.py

```diff
@@ -597,3 +598,91 @@ def get_aggregate_id(aggregate):
     id_string = ",".join(job.get_id() for job in aggregate)
     hash_ = md5(id_string.encode("utf-8")).hexdigest()
     return f"agg-{hash_}"
+
+
+class _AggregatesCursor(Collection):
+    """Abstract class defining iterators over aggregates stored in a FlowProject.
+
+    Parameters
+    ----------
+    project : :class:`~.FlowProject`
+        A FlowProject whose jobs are aggregated.
+
+    """
+
+    @abstractmethod
```

Comment:
We want the `_AggregatesCursor` parent class to define an interface. By deriving from the abstract base class `Collection` and marking the constructor `__init__` and equality operator `__eq__` as abstract, we can create subclasses `_AggregateStoreCursor` and `_JobAggregateCursor` that are required to implement `__init__`, `__eq__`, `__iter__`, `__len__`, and `__contains__`. Specifically we mark the constructor as abstract because we don't want users to be able to construct the base class `_AggregatesCursor` - the constructor behavior should be implemented by the specific cursor classes derived from this abstract base class.

Here's an article explaining the purpose and usage of abstract classes: https://www.geeksforgeeks.org/abstract-classes-in-python/

Reply:
This doesn't make sense to me. I agree that `_AggregatesCursor` should be an abstract class that can't be instantiated, but that is already automatically true by virtue of inheriting from `Collection`, which declares `__contains__`, `__iter__`, and `__len__` abstract. We also need `__eq__` to be defined by subclasses. Declaring the _constructor_ abstract, on the other hand, prevents a user from defining a subclass with a trivial constructor. While the subclasses that we define happen to require stateful constructors, that doesn't make sense as a requirement.

Reply:
Okay. I removed the abstract constructor in 51f9aef31d956bfc7a26e25e559d7bd3c18156a5.